util/Pos/Util/Orphans.hs

```diff
@@ -233,3 +231,15 @@ type instance Distribution (Ether.TaggedTrans tag t m) = Distribution m
 type instance SharedExclusiveT (Ether.TaggedTrans tag t m) = SharedExclusiveT m
 type instance Gauge (Ether.TaggedTrans tag t m) = Gauge m
 type instance ChannelT (Ether.TaggedTrans tag t m) = ChannelT m
+
+instance MonadUnliftIO (t m) => MonadUnliftIO (Ether.TaggedTrans tag t m) where
+    {-# INLINE askUnliftIO #-}
+    askUnliftIO =
+        Ether.TaggedTrans $
+        withUnliftIO $ \u ->
+        return (UnliftIO (unliftIO u . \case Ether.TaggedTrans trans -> trans))
```

Comment:
I can see why this typechecks, but I can't be sure it is _correct_.

Reply:
Copied from https://hackage.haskell.org/package/unliftio-core-0.1.1.0/docs/src/Control-Monad-IO-Unlift.html#line-91